src/main.cpp

```diff
@@ -739,6 +760,7 @@ QString UsageArguments ( char** argv )
            "                        supported for headless server mode)\n"
            "  -n, --nogui           disable GUI\n"
            "  -p, --port            set the local port number\n"
+           "      --jsonrpcport     enable JSON-RPC, set port number\n"
```

Comment:
It's maybe worth making it clear _here_ that this will be TCP, rather than UDP, just so people put the right setting in their firewalls, if they're exposing their server.  (Hopefully anyone doing that would realise, though.)

Reply:
I will make it clear that this is a TCP port.

It is worth noting that the current implementation of the RPC **is hardcoded to listen on 127.0.0.1 only,** so I don’t think people would need to put this port on their firewall.

> OK, the main thing I'd like is for it to be possible for any method that can change state to be protected, so someone could expose those methods that retrieve state whilst not allowing control of a server.

With the TCP server listening to `localhost` only, people who want to remotely control the server would `ssh` into the machine, and then use `LocalForward` to forward the port to their workstation.

Do you think this would offer enough protection?

Here’s how I am envisioning it: `--jsonrpcport` provides a bare-bones IPC mechanism to programmatically interact with a Jamulus client/server (not meant to be exposed). This allows better APIs to be written on top as needed.

```
     +------------+       - Written by others.
     | API server |       - Can perform authn/authz.
     +------------+       - Can use RESTful, GraphQL, gRPC, WebSockets, etc.
            |
            V
  +------------------+    - Listen on localhost only.
  | Jamulus JSON-RPC |    - Bare-bones newline-delimited JSON-RPC.
  +------------------+    - No encryption or access control.
```

For example, [this `visualizer-server.js`](https://github.com/dtinth/jamcaster/blob/eeff891345d9fb7bdac63778b015e017d1f4c328/src/visualizer-server.js) exposes the connected clients and their sound level through [HTTP server-sent events](https://en.wikipedia.org/wiki/Server-sent_events), so that it can be [visualized on a web page](https://user-images.githubusercontent.com/193136/131781233-75641c1a-fa81-47d5-b6fe-3eeaba61cc3a.gif).

Reply:
Ah, I hadn't noticed the limitation to 127.0.0.1 (or hadn't taken it in - I'd read it...).  I'd also make that clear in the help that it's a `local TCP port` that's needed.

Reply:
Done — Help text updated

e50d4595a5e06a0d587817b58732db54ffe1ce1b